### PR TITLE
MUL 243: Ethereum transaction serialization and signing.

### PR DESCRIPTION
**Pull request by Enmk**
Implemented signing and serializating transactions;
Added enum with predefined chain ids;
Fixed test case data to match serialized transaction data.

### CODE REVIEW

File: multy_transaction/internal/ethereum_transaction.cpp [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+    serialize_to_stream(data_stream, *m_chain_id > 0 ? SERIALIZE_WITH_CHAIN_ID : SERIALIZE);
+
+    m_signature.reset(new EthereumTransactionSignature);
+    m_signature->set_signature(m_account.get_private_key()->sign(
+            data_stream.get_content()));
+}
+
+Amount EthereumTransaction::estimate_total_fee(size_t sources_count, size_t destinations_count) const
+{
+    return Amount(1);
```
Comment (PashaKlybik) [inline multy_transaction/internal/ethereum_transaction.cpp]: Why estimate_total_fee always return(1)?

File: multy_transaction/internal/ethereum_transaction.cpp [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+const uint8_t RLP_DATA_IMM_LEN_COUNT = RLP_LIST_START - RLP_DATA_IMM_LEN_START - RLP_MAX_LENGTH_BYTES;
+const uint8_t RLP_DATA_IND_LEN_ZERO = RLP_DATA_IMM_LEN_START + RLP_DATA_IMM_LEN_COUNT - 1;
+const uint8_t RLP_LIST_IMM_LEN_COUNT = 256 - RLP_LIST_START - RLP_MAX_LENGTH_BYTES;
+const uint8_t RLP_LIST_IND_LEN_ZERO = RLP_LIST_START + RLP_LIST_IMM_LEN_COUNT - 1;
+
+const int PRE_EIP155_CHAIN_ID = -4;
+
+#define LV(x) (#x " : ") << std::hex << static_cast<int>(x)
+
+const struct Foo
```
Comment (PashaKlybik) [inline multy_transaction/internal/ethereum_transaction.cpp]: Where this class is used?

File: multy_transaction/internal/ethereum_transaction.cpp [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+    const size_t PAYLOAD_NONZERO_BYTE_GAS = 68;
+    const size_t TX_BASE_GAS = 21000;
+
+    const BinaryData& data = m_payload.is_set() ? **m_payload : BinaryData{nullptr, 0};
+    const size_t zero_bytes_count = std::count(data.data, data.data + data.len, 0);
+
+    const size_t estimated_gas = TX_BASE_GAS
+            + zero_bytes_count * PAYLOAD_ZERO_BYTE_GAS
+            + (data.len - zero_bytes_count) * PAYLOAD_NONZERO_BYTE_GAS;
+    return estimated_gas * *m_fee->gas_price;
```
Comment (PashaKlybik) [inline multy_transaction/internal/ethereum_transaction.cpp]: We haven't Amount constructor from size_t

**Review by PashaKlybik — COMMENTED** [review]
Also add in multy_test/CMakeList.txt and multy_transaction/CMakeList.txt on target_link_libraries "libwally-core"

**Comment by Enmk** [comment]
Should be Ok now.

**Comment by PashaKlybik** [comment]
All OK!

**Comment by PashaKlybik** [comment]
Ok!

**Pull request merged**